[김지연] 스프린트1,2

dani784601 opened the pull request:
## 요구사항

### 기본
#### 스프린트 미션1
- [x] PC사이즈만 고려해 주어진 디자인을 구현합니다.
- [x] HTML, CSS 파일을 Netlify로 배포해 주세요.
  - <a href="https://jocular-mermaid-f07db6.netlify.app/">https://jocular-mermaid-f07db6.netlify.app/</a>
- [x] “로그인”버튼 클릭 시 로그인 페이지(‘/login)로 이동합니다
- [x] “구경하러가기”버튼 클릭 시(’/items’)로 이동합니다.
- [x] 아래로 스크롤 해도 “판다 마켓” 로고와 “로그인” 버튼이 있는 상단 네비게이션 바가 최상단에 고정되게 해주세요.
- [x] 클릭으로 기능이 동작해야 하는 경우, 사용자가 클릭할 수 있는 요소임을 알 수 있도록 CSS 속성 cursor: pointer 로 설정합니다.

<b>화면의 너비가 1920px 이상이면</b>
  - [x] 하늘색 배경색은 너비를 꽉 채우도록 채워지고, 내부 요소들의 위치는 고정되고, 여백만 커지도록 합니다.
  - [x] 내부에 있는 요소간 동일한 간격을 유지하며 가운데 정렬해야 합니다.

<b>화면의 너비가 1920px 보다 작아질 때</b>
  - [x] “판다마켓” 로고의 왼쪽 여백 200px, “로그인" 버튼의 오른쪽 여백 200px이 유지되고, 화면의 너비가 작아질수록 두 요소간 거리가 가까워지도록 설정합니다.
---
#### 스프린트 미션2
- 로그인 페이지, 회원가입 페이지 공통
  - [x] “판다마켓" 로고 클릭 시 루트 페이지(“/”)로 이동합니다.
  - [x] 로그인 페이지, 회원가입 페이지 모두 로고 위 상단 여백이 동일합니다.
  - [x] SNS 아이콘들은 클릭 시 각각 “https://www.google.com/”, “https://www.kakaocorp.com/page/” 으로 이동합니다.
  - [x] input 요소에 focus in 일 때, 테두리 색상은 ##3692FF입니다.
  - [x] input 요소에 focus out 일 때, 테두리는 없습니다.

- 회원가입 페이지
  - [x] “로그인”버튼 클릭 시 “/login” 페이지로 이동합니다.

- 로그인 페이지
  - [x] “회원가입”버튼 클릭 시 “/signup” 페이지로 이동합니다.

### 심화
#### 스프린트 미션1
- [x] reset.css를 설정해주세요.
---
#### 스프린트 미션2
- [ ] 비밀번호, 비밀번호 확인 input 요소 오른쪽에 비밀번호를 확인할 수 있는 눈 모양 - 아이콘을 추가합니다.
- [x] palette에 있는 color값들을 css 변수로 등록해서 사용합니다.
- [x] 구글 애널리틱스로 방문자 수 확인하기할 수 있도록 설정합니다.

## 주요 변경사항
- favicon 설정
- 메인 페이지 추가(스프린트 미션1)
- 로그인, 회원가입 페이지 추가(스프린트 미션2)

## 스크린샷
-

## 멘토에게
-

sprint-edu committed:
docs: update branch 구조 in readme

sprint-edu committed:
docs: update README.md

sprint-edu committed:
docs: update readme

프론트엔드/백엔드 레포 분리

sprint-edu committed:
docs: update readme

프론트엔드/백엔드 레포 분리

sprint-edu committed:
feat: add: auto-labeling and assignee for PRs

dani784601 committed:
feat: Add login and signup page styles

- Add login.css and signup.css files with imported common.css and components/form.css
- Add common.css with imported base/reset.css, base/color.css, base/font.css, base/layout.css, and base/button.css
- Add biome.json file with JavaScript and CSS configurations
- Add qna.html, items.html, and privacy.html pages with basic HTML structure
- Add components/form.css with styles for input fields
- Add components/footer.css with styles for the footer section
- Add social media icons (facebook-logo.svg and youtube-logo.svg)
- Add components/header.css with styles for the header section
- Add base/button.css with styles for buttons
- Add components/card.css with styles for card components
- Add base/reset.css with basic CSS reset styles
- Add base/color.css with color variables

dani784601 committed:
style: Update items.html link path

orlein commented on biome.json:
이 파일은 `Biome` 이라는 서비스에서 온 것 같습니다. 
https://biomejs.dev/guides/configure-biome/ 참조하시고 json schema 넣어주시면 더욱 좋습니다.

dani784601 replied:
적용완료

orlein commented on index.html:
```diff
@@ -0,0 +1,119 @@
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>판다 마켓</title>
```
아주 사소합니다만, 지시사항은 `판다마켓`입니다. 
지금같은 퍼블리싱 측면에서는 틀려도 괜찮긴 합니다만, 프론트엔드 개발에서는 꽤 치명적인 부분이 될 수 있습니다. 서비스 이름이 데이터베이스부터 내려오는 고정적인 값으로 보여줄 필요가 있습니다. '판다마켓' 을 검색할때 딱 하나만 빼놓고 검색된다면 생각보다 귀찮은 문제가 생길수도 있습니다.
지금은 괜찮습니다! 프론트개발 하신다면, 다음엔 이런 공통적인 부분은 전역변수로 빼보세요.

dani784601 replied:
적용완료

orlein commented on index.html:
```diff
@@ -0,0 +1,119 @@
+  <section class="bg-main container">
+    <div class="content flex justify-center">
+      <div class="flex flex-row flex-1">
+        <h1 class="text-3xl sm-text-4xl font-bold gray700">일상의 모든 물건을 <br class="md-none">거래해 보세요</h1>
```
클래스이름이나, 다른 부분에서도 프론트엔드 경험을 잘 쌓아오신것이 드러나고있습니다. 훌륭합니다. tailwind 클래스이름이 적절히 사용된것이 인상깊네요.
하지만 개발에서는 `코드 재사용` 이라는 관점에서 엔지니어링이 시작됩니다. 특히나 지금같은 경우, `일상의 모든 물건을` <br /> `거래해 보세요` 라고 표현하신다면, 다른 어떤 컴포넌트 재사용 상황에서도 끊어져있게 됩니다. 
물론 이 section을 컴포넌트화시켜도 저 문구는 끊어져있을 가능성이 매우 높고, 현재의 스프린트 미션에서는 끊어지는게 맞습니다. 
틀렸기때문에 리뷰 메시지를 달아놓는것이 아니고, 오히려 지시사항을 아주 잘 이행하셨는데,
한번 저 br을 없앴을 때도 의도한 word break가 일어난다면, br을 없애보는걸 권장드립니다.

orlein commented on index.html:
```diff
@@ -0,0 +1,119 @@
+        <a href="pages/items.html" class="btn btn-block bg-primary">구경하러 가기</a>
+      </div>
+      <div class="flex flex-2">
+        <img src="images/home/hero-image.png" width="746" height="340"
+          alt="a painting of a penguin and a cartoon character ">
```
`alt` 아주 좋습니다.

orlein commented on index.html:
```diff
@@ -0,0 +1,119 @@
+    <ul class="flex space-between align-center">
+      <li class="gray400">©codeit - 2024</li>
+      <li class="gray200 flex gap-30">
+        <a href="/privacy.html">Privacy Policy</a>
+        <a href="/faq.html">FAQ</a>
```
링크가 동작하지 않습니다. 아마 단순한 실수일것같네요. 수정 부탁드립니다.

dani784601 replied:
수정완료

orlein commented on pages/items.html:
```diff
@@ -0,0 +1,14 @@
+  <title>Document</title>
+</head>
+
+<body>
+  <h1>items</h1>
+</body>
```
지시사항이 `빈 페이지` 라고 해서 정말로 빈 페이지를 놓지 않고 적절한 문구를 넣어두신것 아주 좋습니다.

orlein commented on index.html:
```diff
@@ -0,0 +1,119 @@
+  <header class="w-full">
+    <div class="mx-auto flex-1 header-body flex space-between align-center">
+      <a href="/" class="shrink-0">
+        <img aria-label="판다마켓 홈버튼" class="logo" src="images/logo/panda-market-logo.png" width="153" height="51" alt="">
+        <img aria-label="판다마켓 홈버튼(모바일)" class="mb-logo" src="images/logo/mb_panda-market-logo.png" width="103"
+          height="51" alt="">
```
`aria-label` 넣어두신것 너무 좋습니다.

orlein commented on pages/items.html:
```diff
@@ -0,0 +1,14 @@
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Document</title>
```
지시사항은 없었지만, 맥락을 고려하면 title은 `판다마켓`인 편이 좋다고 봅니다.

dani784601 replied:
적용완료(전역변수 사용)

orlein commented on styles/base/variables.css:
css variable 사용 아주 좋습니다. 혹시 dev mode에서 figma variables를 가져오신건가요? 제가 권한이 없다보니 이 부분은 체크가 안되네요.

dani784601 replied:
dev 모드 사용한 건 아니고 코파일럿으로 생성했습니다ㅎㅎ;

orlein commented on styles/base/variables.css:
```diff
@@ -0,0 +1,60 @@
+:root {
+  /* Font Sizes */
+  --font-size-4xl: 40px;
+  --line-height-4xl: 56px;
+  --font-size-3xl: 32px;
+  --line-height-3xl: 42px;
+  --font-size-2xl: 24px;
+  --line-height-2xl: 32px;
+  --font-size-xl: 20px;
+  --line-height-xl: 32px;
+  --font-size-2lg: 18px;
+  --line-height-2lg: 26px;
+  --font-size-lg: 16px;
+  --line-height-lg: 26px;
+  --font-size-md: 14px;
+  --line-height-md: 24px;
+  --font-size-sm: 13px;
+  --line-height-sm: 22px;
+  --font-size-xs: 12px;
+  --line-height-xs: 18px;
```
현재는 `PC 해상도만 고려` 하기 때문에 px도 괜찮습니다만 나중에는 `rem`을 사용해보세요.
https://velog.io/@codns1223/CSS-px-em-rem-%EC%B0%A8%EC%9D%B4
읽을만한 문서 추천드립니다

orlein commented on styles/components/card.css:
```diff
@@ -0,0 +1,58 @@
+
+.card-list {
+  max-width: 988px;
```
다른 css variable처럼 breakpoint도 variable로 처리해보시는걸 권장드려요. 추가적인 실수를 방지해줍니다.

dani784601 replied:
미디어 쿼리에 적용해보려고 했는데, 아직 미디어 쿼리에서는 변수 사용이 안 된다고 하네요ㅠㅠ e.g @media (max-width: 375px) { .. (적용X)

orlein commented on styles/components/card.css:
```diff
@@ -0,0 +1,58 @@
+  .card-list {
+    margin-bottom: 56px;
+  }
+  .card-img {
+    order: 0;
+    img {
+      display: flex;
+      width: 100%;
+    }
+  }
```
`css nesting`이 sass같은데서는 익숙하지만 막상 pure css에서는 얼마 소개되지 않은 기능인데 잘 사용해주셨습니다.

orlein commented on styles/components/footer.css:
```diff
@@ -0,0 +1,28 @@
+
+footer {
+  min-height: 160px;
+  padding: var(--padding-32) var(--padding-200);
```
padding pixel값을 css variable로 처리한것 아주 좋습니다. 다음에는 한번 semantic하게, 의미가 있는 변수명을 지정해보세요.

orlein reviewed (commented):
전체적으로 훌륭합니다. 다른분들과 협업한다면 충분히 리드가 가능하시겠네요.

dani784601 committed:
Fix: Update biome.json formatting options($schema)

dani784601 committed:
feat: 사이트 이름 상수값으로 설정, 잘못된 링크 수정

dani784601 committed:
feat: 폰트 굵기별 설정 세분화 및 woff2 파일 형식으로 변경

dani784601 committed:
fix: 잘못된 파일 경로 수정

dani784601 committed:
style: normalize.css 추가, css 브레이크 포인트 변수 설정

dani784601 committed:
style: normalize.css 대신 tailwindcss의 preflight.css 사용

dani784601 committed:
feat: GA 태그 추가

dani784601 commented:
안녕하세요, @orlein 님

리뷰해 주셔서 감사합니다. 제안해주신 사항들을 반영하여 다음과 같이 수정하였습니다:

1. `CONSTANS.js`로 `title` 값 분리
2. 잘못된 링크 및 경로 수정
3. CSS 변수에 breakpoint 추가

추가로, 다음과 같은 개선 사항을 적용하였습니다:

1. 페이지별 GA 태그 추가
2. 세부적인 폰트 스타일 지정
3. 직접 작성한 `reset.css` 대신 TailwindCSS의 `preflight.css` 파일 적용

혹시 추가로 검토해 주실 사항이 있으면 말씀해 주세요. 감사합니다!🙂

orlein commented on CONSTANS.js:
현재로써는 가장 좋은 방법이긴 한데, 나중에는 별로 좋은방법이 아니게 됩니다. 지금 당장은 설명하기보다 나중에 설명드릴께요.

dani784601 replied:
✅

orlein commented on styles/base/font.css:
이런 방법으로 구현하시는것도 좋은 방법입니다만 가장 좋은건 Pretendard의 공식 레포지토리를 참조하는겁니다
https://github.com/orioncactus/pretendard
가변글꼴 사용이 가능하다면 간결하게 가변글꼴로 처리할 수도 있습니다

dani784601 replied:
가변글꼴에 대해 알려주셔서 감사합니다! 관련해서 [구글: 가변 폰트의 놀라운 활용법 | 요즘IT](https://yozm.wishket.com/magazine/detail/1239/) 이런 기사를 읽어봤는데, 도움이 됐습니다.
그런데 가변글꼴은 일반글꼴과 동일하게 사용하면 되는 건가요? 혹시 호완성 이슈 같은 주의점이 있어서 사용자에게 일반 글꼴과 가변 글꼴을 모두 제공해야 하는지 궁금합니다.

orlein replied:
https://caniuse.com/variable-fonts 호환성 이슈는 IE와 Opera Mini를 제외하면 없다고 보셔도 되는데 사실상 **기술적인 이슈는 없다시피 하다**고 보는게 타당합니다. 사실 정말로 주의해야할 이슈는 디자인적 이슈인데... 사실 그런 이슈를 체크하는 역량은 제게 없습니다 ^^; 되도록 디자이너분께 이렇게 표현되는게 맞냐고 크로스체크 해봐야죠.

orlein commented on styles/base/layout.css:
```diff
@@ -6,7 +6,7 @@
   display: flex;
 }
-.flex-row {
+.flex-col {
   flex-direction: column;
 }
```
제가 리뷰를 놓친 부분인데 잡아주셨네요 좋습니다 ^^;

orlein commented on styles/base/variables.css:
```diff
@@ -57,4 +57,9 @@
   --border-radius-lg: 8px;
   --border-radius-md: 6px;
   --border-radius-sm: 4px;
+
+  /* Break Points */
+  --breakpoint-mb: 375px;
+  --breakpoint-tb: 744px;
+  --breakpoint-pc: 1200px;
```
👍

dani784601 committed:
ㄹeat: Add missing title tag and update image alt text

- Add missing title tag to the HTML document
- Update image alt text to provide more descriptive information

Refactor the HTML code to improve accessibility and SEO.

dani784601 committed:
Feat: Update copyright year dynamically

dani784601 committed:
Merge branch 'codeit-sprint-fullstack:main' into basic-김지연-sprint2

dani784601 committed:
style: padding 단위를 0.25rem으로 변경

dani784601 committed:
style: login 페이지 여백 조정

dani784601 committed:
style: Adjust login page layout and spacing

dani784601 committed:
style: gap-*, p-* class속성도 css variable로 정의

dani784601 committed:
style: Adjust login page layout and spacing

dani784601 committed:
style: Remove unused CSS imports and adjust login and sign-up page layout

dani784601 committed:
style: Adjust layout and spacing in main page

dani784601 committed:
style: Adjust layout and spacing in main page

dani784601 committed:
style: Adjust layout and spacing in sign-in, sign-up page

dani784601 committed:
style: fix typo

dani784601 committed:
feat: Add 404 page and redirect file

dani784601 committed:
style: Adjust layout and spacing in header, footer

orlein commented on .github/workflows/auto-label-assign.yml:
와우 github workflow까지... 아예 매운맛을 기본장착해두려고 하시는군요 무섭습니다 ㅋㅋㅋㅋ

dani784601 replied:
이건 코드잇 설정입니다~ㅎㅎ;

orlein commented on 404.html:
에러페이지는 사용자경험에서 중요한 역할을 합니다. 
지금은 (백엔드 지원이 없으니) 좀 어렵겠지만, 나중에는 `이 에러가 왜 떴는지` 에 대해 충분한 설명을 하고나서 `유저가 우리 서비스로부터 이탈하지 않게` 하는 버튼이나 액션 등을 제공해줘야겠습니다.

orlein commented on CONSTANS.js:
Redux나 Zustand같은 `global state management`가 없기때문에 사용하고있는 것으로 생각됩니다. 좋습니다.
만약에 나중에 그런걸 사용하실 수 있게 되면 사용하시고, 혹시 나중에 현업에서 기술적으로 혹은 다른 이유로 서버사이드 지원을 전혀 바랄 수 없다면, web server로부터의 cookie나 web browser의 local storage 등을 사용하셔도 좋습니다.

그리고 좀 사소합니다만 copyright에서 그냥 since 써놓는편이 더 낫습니다 ㅎㅎ

dani784601 committed:
feat: 스프린트 미션 3 기본 요구사항 - 로그인 페이지 공통 요구사항(에러메세지, 로그인 버튼 동작)

orlein reviewed (commented):
👍 이어질 스프린트에서도 코드리뷰 기대하겠습니다.

dani784601 committed:
feat: abort controller를 사용하여 window unload 이벤트 발생시 등록했던 event handler 제거

dani784601 committed:
feat: 스프린트 미션 3 심화 요구사항 - 오류 메시지 모달 구현

dani784601 committed:
feat: 스프린트미션3 기본요구 사항 - 회원가입 페이지(진행중)

orlein reviewed (approved):
👍 (merge 이후에 작업바랍니다)

dani784601 committed:
feat: 스프린트미션3 기본요구 사항 - 회원가입 페이지 유효성 검사, 스프린트 미션2 심회 요구사항 - 비밀번호 확인 아이콘 추가
